tests/unit/Term/TermFallbackTest.php

```diff
@@ -53,11 +54,20 @@ public function testGivenInvalidSourceLanguageCode_constructorThrowsException( $
 
 	public function invalidSourceLanguageCodeProvider() {
 		return array(
-			array( 21 ),
-			array( '' ),
+			'integer' => array( 21 ),
+			'emptyString' => array( '' ),
+			'sameAsRequestedLanguageCode' => array( 'foor' ),
 		);
 	}
 
+	/**
+	 * @expectedException InvalidArgumentException
+	 */
+	public function testGivenInvalidRequestedSameAsActualAndNoTransliteration_constructorThrowsException() {
```

Comment:
> ( 'requested', 'text', 'requested', 'source' ) => exception

This is valid, added an explicit test for it.

> ( 'requested', 'text', 'requested', 'requested' ) => exception

I still need to add this and I should probably move all the exception throwing test variants into one data provider. The rest already have tests.